recipes-devtools/rust/rust.inc

```diff
@@ -315,6 +366,10 @@ def rust_gen_mk_cfg(d, thing):
         '-e', '/^CFG_GCCISH_CXXFLAGS/ s;$; {};'.format(cxxflags_for(d, thing)),
         '-e', '/^CFG_GCCISH_LINK_FLAGS/ s;$; {};'.format(ldflags),
 
+        # The mipsel target specify flags for the mips32. Replace them with
+        # mips32r2 if that's what we use
+        '-e', 's/mips32/mips32r2/g' if is_mips32r2 else '',
```

Comment:
In gcc (and many other command line interfaces with options), later arguments override earlier ones, with the intent of supporting cases like this one. Appending the flag should work fine unless `-mips32` somehow implies something slightly different than a `-march` option.

Another option here is just to have meta-rust generate the full set of cflags, which I have considered in the past. Trying to build on top of the `mk/cfg` stuff instead of generating it completely is a bit funky (but it was quick to get working).